src/bsp/platform/stm32fx_nucleo_wifi/xi_bsp_io_net_stm32fx_nucleo_wifi.c

```diff
@@ -30,15 +31,21 @@ typedef struct _xi_bsp_stm_net_state_t
 {
     xi_data_desc_t* head;
     xi_bsp_stm_wifi_state_t wifi_state;
-
 } xi_bsp_stm_net_state_t;
 
-xi_bsp_stm_net_state_t xi_net_state = {0};
+/* this is local variable */
+static xi_bsp_stm_net_state_t xi_net_state = {NULL, 0};
 
 xi_bsp_io_net_state_t xi_bsp_io_net_create_socket( xi_bsp_socket_t* xi_socket )
 {
+    /* sanity check */
+    assert( NULL == xi_net_state.head );
+
     *xi_socket = 0;
 
+    /* reset the wifi state */
+    xi_net_state.wifi_state = xi_wifi_state_connected;
```

Comment:
@atigyi - if you take a look at the ```xi_net_state.wifi_state``` value at line 37 - the value 0 is xi_wifi_state_connected - so if we reach this function - we assumed already that it's state was connected - It's true that it is misleading - I'll add another state for uninitialised value - this way our code will be more bulletproof - 💥